.github/workflows/release_build.yml

```diff
@@ -0,0 +1,44 @@
+name: Release build
+on:
+  workflow_dispatch:
+  push:
```

Comment:
this is not better or not worse:

from my expierence make sense separate push top branch from release:
- Maybe you want to validate artifacts before release
- Prepare documentation before release on nuget
- Maybe you want to have a dealy with release
- Release candidates
- e.t.c

Reply:
I think there are lots of improvement we can make to the build. I'm not sure if we should tackle those now or in a future PR?

Some options:

* Change versioning to not use tag (use CHANGELOG version or similar?)
* Proper semver adoption
* Release build auto pushes tag
* Release build creates GitHub releases entry
* Release build pushes to nuget
* Doc build to push new web site

I'd prefer not to have version as an input to build workflow as I'd like the version of each commit to be apparent from the repository state. Based on that I think it might be best to keep the current build-on-tag approach until we get a chance to overhaul the whole process.

In other projects I've switched from tags to a `VERSION` file containing this info which IMO works a lot better than the tag-based system we adopted here a long, long time ago. :)

Reply:
Yea, it could be changed in future PR.

Reply:
Also vote for future PR